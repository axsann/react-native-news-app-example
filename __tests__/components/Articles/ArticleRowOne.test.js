// @flow

import 'react-native'
import React from 'react'
import ArticleRowOne from '../../../src/components/Articles/ArticleRowOne.component'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const mockItem = {"title":"あああああああああああああああああああああああああああ","url":"https://example.com","date":"20XX年Y月Z日","imgUrl":"https://placehold.jp/150x150.png","tags":["タグB"]}

describe('ArticleRowOne', () => {
  it('renders correctly', () => {
    const instance = renderer.create(
      <ArticleRowOne article={mockItem}  onPressArticle={()=>{}} />
    )

    expect(instance.toJSON()).toMatchSnapshot()
  })
})
