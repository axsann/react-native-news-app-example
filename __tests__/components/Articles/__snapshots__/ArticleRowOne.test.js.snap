// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleRowOne renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
      "height": 100,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontWeight": "bold",
        }
      }
    >
      あああああああああああああああああああああああああああ
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 10,
          "marginTop": 5,
        }
      }
    >
      20XX年Y月Z日
    </Text>
  </View>
  <Image
    source={
      Object {
        "uri": "https://placehold.jp/150x150.png",
      }
    }
    style={
      Object {
        "flex": 2,
      }
    }
  />
</View>
`;
